src/lib/plugins/placeBlock.js

```diff
@@ -81,43 +93,22 @@ module.exports.player = function (player, serv, { version }) {
     })
 
     if (!blocks[id]) return
-    player.behavior('placeBlock', {
-      direction: directionVector,
-      heldItem: heldItem,
-      id,
-      damage: data,
-      position: placedPosition,
-      reference: referencePosition,
-      playSound: true,
-      sound: 'dig.' + (materialToSound[blocks[id].material] || 'stone')
-    }, ({ position, playSound, sound, id, damage, heldItem }) => {
-      if (playSound) {
-        serv.playSound(sound, player.world, placedPosition.clone().add(new Vec3(0.5, 0.5, 0.5)), {
-          pitch: 0.8
-        })
-      }
 
-      if (player.gameMode === 0) { player.inventory.slots[36 + player.heldItemSlot]-- }
-      var stateId;
-      console.log(mcData.blocksByName[heldItem.name])
-      if (serv.supportFeature("theFlattening")) {
-        if (mcData.blocksByName[heldItem.name].name == 'grass_block')
-          damage = 1;
-        stateId = mcData.blocksByName[heldItem.name].minStateId + damage
-      } else
-        stateId = id << 4 | damage
-      player.setBlock(position, stateId)
-
-      if (id === 63 || id === 68) {
-        player._client.write('open_sign_entity', {
-          location: position
-        })
-      }
-    }, async () => {
-      const id = await player.world.getBlockType(placedPosition)
-      const damage = await player.world.getBlockData(placedPosition)
-      player.sendBlock(placedPosition, id, damage)
+    const sound = 'dig.' + (materialToSound[blocks[id].material] || 'stone')
+    serv.playSound(sound, player.world, placedPosition.offset(0.5, 0.5, 0.5), {
+      pitch: 0.8
     })
+
+    if (player.gameMode === 0) { player.inventory.slots[36 + player.heldItemSlot]-- }
+
+    const stateId = serv.supportFeature('theFlattening') ? (blocks[id].minStateId + data) : (id << 4 | data)
+    player.setBlock(placedPosition, stateId)
+
+    if (id === 63 || id === 68) {
```

Comment:
this seems wrong (hardcoded ids)

Reply:
indeed, this part have nothing to do here anyway, it should be move in signs.js

Reply:
done